router/auth.go

```diff
@@ -84,6 +84,42 @@ func authPassThrough(log *log.Logger, w http.ResponseWriter, req *http.Request,
 
 func authInjectECS(log *log.Logger, w http.ResponseWriter, req *http.Request, target *targetPassThroughAuth) bool {
 	req.SetBasicAuth(target.config.Username, target.config.Password)
+	var didWhat string
```

Comment:
Let's put this whole chunk of code in a new wrapper with a similar signature to authInjectECS.  Probably a good idea to make it part of serviceauth submodule. This will allow re-use in other golang code